app/src/main/java/ch/epfl/cs311/wanderwave/viewmodel/TrackListViewModel.kt

```diff
@@ -19,14 +20,47 @@ class TrackListViewModel
 @Inject
 constructor(
     private val spotifyController: SpotifyController,
+    private val appDatabase: AppDatabase,
     private val repository: TrackRepository
 ) : ViewModel() {
 
   private val _uiState = MutableStateFlow(UiState(loading = true))
   val uiState: StateFlow<UiState> = _uiState
+  private var _currentBeaconId: String? = null
+  var currentBeaconId: String?
+    get() = _currentBeaconId
+    set(value) {
+      _currentBeaconId = value
+      loadTracksBasedOnSource()
+    }
```

Comment:
Why is the currentBeaconId on the TrackListViewModel? The track list screen should not worry about beacons